### DISCUSSION TITLE
create JWT using EC private key

### DISCUSSION THREAD

**Discussion by willyprayogo26** [panva/jose#425] [Q&A]
I generate private key using this command:
`openssl ecparam -out private.key -name secp521r1 -genkey`

but I cannot import EC private key to use in:
```
    const keyFile = fs.readFileSync('./private.key')
    const ecPrivateKey = await jose.importPKCS8(keyFile.toString(), 'ES256')

    const payload = {
          sub: xxx,
          aud: xxx',
          iss: xxx,
          iat: Date.now(),
          exp: 120
    }
    await new jose.SignJWT(payload)
          .setProtectedHeader({ alg: 'ES256', typ: 'JWT', kid: 'xxxx' })
          .sign(ecPrivateKey)
```
could u help me how to solve this?
thanks

**Comment by panva** [comment]
1) your `private.key` file does not just contain the PKCS#8 envelope that [`importPKCS8`](https://github.com/panva/jose/blob/main/docs/functions/key_import.importPKCS8.md) expects. It has other content in it that's not expected.
2) `secp521r1` is not usable with the `ES256` JWS Algorithm. The algorithm name for `secp521r1` curve EC key would `ES512`. see #210

**Reply by willyprayogo26** [reply]
ok I just tried to use `ES512`, but still got an error:
```
ERROR [ExceptionsHandler] "pkcs8" must be PCKS8 formatted string
```
so how to solve this?

**Reply by panva** [reply]
see 1) your key is SEC1, not PKCS8
